LibertyDSNP/contracts
# Registry Contract

Problem
=======
The Registry contract should be implemented to spec: https://spec.projectliberty.io/Identity/Registry
[link to Pivotal Tracker #12345678](https://www.pivotaltracker.com/story/show/177872246)

Solution
========
We implemented the registry contract. In order to test authorization we also added a `TestDelegation` contract. This may eventually be replaced with an Identity contract.

This does not yet contain EIP 712 methods.

Change summary:
---------------
* Introduce `IRegistry` interface.
* Implement `IRegistry.register`method.
* Implement `IRegistry.changeAddress`method.
* Implement `IRegistry.changeHandle`method.
* Introduce `IDelegate` interface (required to authorize)
* Introduce `TestDelegate` and `TestERC165` contracts just for testing.
* Test all methods.

Steps to Verify:
----------------
1. Run tests and ensure they pass.
1. Inspect code?
